File: app/src/main/java/in/mvpstarter/sample/injection/ConfigPersistent.java
package in.mvpstarter.sample.injection;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

import in.mvpstarter.sample.injection.component.ConfigPersistentComponent;

/**
 * A scoping annotation to permit dependencies conform to the life of the
 * {@link ConfigPersistentComponent}
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface ConfigPersistent {
}

File: app/src/main/java/in/mvpstarter/sample/injection/ConfigPersistent.kt
package `in`.mvpstarter.sample.injection


import javax.inject.Scope

import `in`.mvpstarter.sample.injection.component.ConfigPersistentComponent

/**
 * A scoping annotation to permit dependencies conform to the life of the
 * [ConfigPersistentComponent]
 */
@Scope @Retention annotation class ConfigPersistent